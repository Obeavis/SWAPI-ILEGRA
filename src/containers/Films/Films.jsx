import React from "react";
import PropTypes from "prop-types";
import { Container, Row } from "react-bootstrap";
import Film from "../../components/Film/Film";
import "./Films.scss";

function loadImage(idMovie) {
	try {
		return require(`../../assets/images/films/episode-${idMovie}.jpg`)
	} catch (error) {
		return require(`../../assets/images/no-image.png`)
	}
}

const Films = (props) => (
	<Container className="film-select">
		<Row>
			{props.films.map((film, i) => {
				return (
					<Film film={film} idMovie={i+1} image={loadImage(film.episodeID)} key={i}/>
				)
			})}
		</Row>
	</Container>
)

Films.propTypes = {
	Films: PropTypes.object
};
export default Films