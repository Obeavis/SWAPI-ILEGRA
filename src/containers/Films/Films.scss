.film-select {
  .box-films {
    display: flex;
    border-radius: 15px;

    img {
      border-radius: 15px;
      margin-left: auto;
      margin-right: auto;
    }
  }
}
