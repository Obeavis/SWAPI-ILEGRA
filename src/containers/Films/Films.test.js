import React from "react";
import { MemoryRouter } from 'react-router-dom';
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import Films from "./Films";

configure({ adapter: new Adapter() });

describe("Films container", () => {
	const films = [
		{ episodeID: 4, title: "A New Hope", releaseDate: "1977-05-25" },
		{ episodeID: 5, title: "The Empire Strikes Back", releaseDate: "1980-05-17" },
		{ episodeID: 6, title: "Return of the Jedi", releaseDate: "1983-05-25" },
		{ episodeID: 1, title: "The Phantom Menace", releaseDate: "1999-05-19" },
		{ episodeID: 2, title: "Attack of the Clones", releaseDate: "2002-05-16" },
		{ episodeID: 3, title: "Revenge of the Sith", releaseDate: "2005-05-19" },
		{ episodeID: 7, title: "The Force Awakens", releaseDate: "2015-12-11" }
	];

	const component = (
		<MemoryRouter>
			<Films films={films} />
		</MemoryRouter>
	);

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

	it("Renders all the 7 films", () => {
		const wrapper = mount(component);
		const quantity = wrapper.find('Film');

		expect(quantity).toHaveLength(7);
	});
});
