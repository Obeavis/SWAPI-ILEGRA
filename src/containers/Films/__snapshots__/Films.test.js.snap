// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films container matches the snapshot 1`] = `
<div
  className="film-select container"
>
  <div
    className="row"
  >
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="A New Hope"
          className="img-fluid"
          src="episode-4.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            A New Hope
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"1977-05-25T00:00:00.000Z"}
            >
              May 24, 1977
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="The Empire Strikes Back"
          className="img-fluid"
          src="episode-5.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            The Empire Strikes Back
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"1980-05-17T00:00:00.000Z"}
            >
              May 16, 1980
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="Return of the Jedi"
          className="img-fluid"
          src="episode-6.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            Return of the Jedi
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"1983-05-25T00:00:00.000Z"}
            >
              May 24, 1983
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="The Phantom Menace"
          className="img-fluid"
          src="episode-1.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            The Phantom Menace
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"1999-05-19T00:00:00.000Z"}
            >
              May 18, 1999
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="Attack of the Clones"
          className="img-fluid"
          src="episode-2.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            Attack of the Clones
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"2002-05-16T00:00:00.000Z"}
            >
              May 15, 2002
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="Revenge of the Sith"
          className="img-fluid"
          src="episode-3.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            Revenge of the Sith
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"2005-05-19T00:00:00.000Z"}
            >
              May 18, 2005
            </time>
          </span>
        </span>
      </div>
    </div>
    <div
      className="mb-5 col-xl-4 col-lg-6"
      onClick={[Function]}
    >
      <div
        className="box-films box"
      >
        <img
          alt="The Force Awakens"
          className="img-fluid"
          src="episode-7.jpg"
        />
        <span
          className="span-box"
        >
          <span
            className="thumb-title text-bold text-center"
          >
            The Force Awakens
          </span>
          <span
            className="thumb-date text-1 text-bold"
          >
            <time
              dateTime={"2015-12-11T00:00:00.000Z"}
            >
              December 10, 2015
            </time>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
